[Good First Issue]: Fix Windows Debug build break in `FeeEstimateQuery.cc` by including the proto header first

### 🆕🐥 First-Time Friendly

This issue is especially welcoming for people who are new to contributing to the **Hiero C++ SDK**.

We know that opening your first pull request can feel like a big step. Issues labeled **Good First Issue** are designed to make that experience easier, clearer, and more comfortable.

No prior knowledge of Hiero, Hedera, or distributed ledger technology is required - just a basic familiarity with C++ and Git is more than enough to get started.

> [!IMPORTANT]
> ### 📋 About Good First Issues
>
> Good First Issues are designed to make getting started as smooth and stress-free as possible.
>
> They usually focus on:
> - Small, clearly scoped changes
> - Straightforward updates to existing code or docs
> - Simple refactors or clarity improvements
>
> Other kinds of contributions — like larger features, deeper technical changes, or design-focused work — are just as valuable and often use the beginner, intermediate, or advanced labels.

### 👾 Description of the Issue

The `Build (Windows)` job on `main` is broken in Debug configuration when compiling [src/sdk/main/src/FeeEstimateQuery.cc](../../src/sdk/main/src/FeeEstimateQuery.cc) (introduced in PR #1594, HIP-1261). MSVC emits 100+ cascading errors inside the proto-generated `basic_types.pb.h` and stops with `error C1003: error count exceeds 100; stopping compilation`.

The translation unit includes [src/sdk/main/include/impl/HttpClient.h](../../src/sdk/main/include/impl/HttpClient.h) (which transitively pulls in `<httplib.h>` → `<windows.h>` on MSVC) **before** `<services/transaction.pb.h>`. The Windows headers `#define` a number of single-word identifiers (`DELETE`, `OUT`, `IN`, `interface`, etc.) that collide with proto-generated enum members and class names, so the proto header is tokenized incorrectly.

Linux and macOS builds pass — this is purely a Windows include-order bug.

The repository already knows about this and applies an established workaround in [src/sdk/main/src/AccountId.cc:2-3](../../src/sdk/main/src/AccountId.cc#L2-L3):

```cpp
// Windows build requires this to be included first for some reason.
#include <services/basic_types.pb.h> // NOLINT
```

`AccountId.cc`, `RegisteredNodeAddressBookQuery.cc`, and `impl/MirrorNodeGateway.cc` all include both `impl/HttpClient.h` and proto headers and build cleanly on Windows because they apply this trick. `FeeEstimateQuery.cc` does not.

Relevant files:

```
src/sdk/main/src/FeeEstimateQuery.cc
```

### 🔁 Steps to Reproduce

1. Check out `main` at the v0.55.0 release commit (`2ae0526`) or later.
2. Configure and build for Windows Debug:
   ```bash
   cmake --preset windows-x64-debug
   cmake --build -j 6 --preset windows-x64-debug --config Debug
   ```
3. Compilation of `FeeEstimateQuery.cc` fails with errors in `build/windows-x64-debug/proto/services/basic_types.pb.h`.

The bug is also reproducible directly in CI on every Windows job run on `main`. See the failing run: [actions/runs/25569708870/job/75062024315](https://github.com/hiero-ledger/hiero-sdk-cpp/actions/runs/25569708870/job/75062024315).

### ✅ Expected Behavior

`FeeEstimateQuery.cc` compiles cleanly on Windows in both Debug and Release configurations, matching the behavior of every other httplib-using TU in the SDK.

### ❌ Actual Behavior

Compilation halts with 100+ errors in `basic_types.pb.h`, beginning at line 306:

```
basic_types.pb.h(306,3): error C2059: syntax error: 'constant'
basic_types.pb.h(316,11): error C4430: missing type specifier - int assumed.
basic_types.pb.h(316,11): error C2737: 'TokenSupplyType': constexpr object must be initialized
basic_types.pb.h(317,11): error C2086: 'const int TokenSupplyType': redefinition
basic_types.pb.h(349,11): error C2737: 'TokenKeyValidation': constexpr object must be initialized
basic_types.pb.h(383,11): error C2737: 'TokenFreezeStatus': constexpr object must be initialized
...
basic_types.pb.h(11284,27): error C2270: 'shardnum': modifiers not allowed on nonmember functions
basic_types.pb.h(11286,10): error C3861: '_internal_shardnum': identifier not found
basic_types.pb.h(11288,22): error C2955: 'ShardID': use of class template requires template argument list
basic_types.pb.h(11288,22): error C1003: error count exceeds 100; stopping compilation
```

The job fails with `Process completed with exit code 1` and the artifact `hapi-library-windows-...` is never produced.

### 🌐 Environment

- SDK version: `main` at commit `2ae0526` (v0.55.0) and later
- Operating system: Windows (`windows-2025` GitHub-hosted runner)
- Compiler and version: MSVC (Visual Studio 2022 toolchain bundled with `windows-2025`)
- Hiero network: N/A — this is a build-time failure
- Configuration: Debug (Release likely affected too but the job aborts before reaching it)

### ✔️ Acceptance Criteria

- [ ] The described behavior is no longer reproducible: `Build (Windows)` passes for both Debug and Release on a PR built from `main`.
- [ ] A regression test is added that fails before the fix and passes after — for this bug, the existing `Build (Windows)` CI job is the regression test (it currently fails on every `main` push and will pass once the include order is fixed).
- [ ] No unrelated behavior or API changes are introduced. In particular, no edits to the public header [src/sdk/main/include/FeeEstimateQuery.h](../../src/sdk/main/include/FeeEstimateQuery.h) and no reordering of unrelated includes.
- [ ] All existing tests continue to pass on Linux and macOS.
- [ ] The fix uses the same `// Windows build requires this to be included first for some reason.` + `// NOLINT` comment-and-include pattern used in [src/sdk/main/src/AccountId.cc:2-3](../../src/sdk/main/src/AccountId.cc#L2-L3), so the workaround stays greppable across the codebase.

### 🤔 Additional Information

**Suggested fix.** Add `#include <services/basic_types.pb.h>` as the very first include in `FeeEstimateQuery.cc`, mirroring `AccountId.cc`. The current top of the file:

```cpp
// SPDX-License-Identifier: Apache-2.0
#include "FeeEstimateQuery.h"
#include "Client.h"
#include "FileAppendTransaction.h"
#include "TopicMessageSubmitTransaction.h"
#include "exceptions/IllegalStateException.h"
#include "impl/HttpClient.h"
#include "impl/MirrorNetwork.h"
```

Should become:

```cpp
// SPDX-License-Identifier: Apache-2.0
// Windows build requires this to be included first for some reason.
#include <services/basic_types.pb.h> // NOLINT

#include "FeeEstimateQuery.h"
#include "Client.h"
#include "FileAppendTransaction.h"
#include "TopicMessageSubmitTransaction.h"
#include "exceptions/IllegalStateException.h"
#include "impl/HttpClient.h"
#include "impl/MirrorNetwork.h"
```

After editing, run `clang-format-17 -i src/sdk/main/src/FeeEstimateQuery.cc` (the `// NOLINT` keeps the new include from being re-sorted).

**Why this works.** `<httplib.h>` drags in `<windows.h>` on MSVC, which `#define`s identifiers that collide with proto-generated enum values. Including the proto header first ensures the proto types are fully declared before any Windows macros are introduced.

**References.**
- Originating PR: [#1594](https://github.com/hiero-ledger/hiero-sdk-cpp/pull/1594) (HIP-1261 fee estimation query).
- Failing run: [actions/runs/25569708870/job/75062024315](https://github.com/hiero-ledger/hiero-sdk-cpp/actions/runs/25569708870/job/75062024315).
- Reference workaround already in the codebase: [src/sdk/main/src/AccountId.cc:2-3](../../src/sdk/main/src/AccountId.cc#L2-L3).
- Other TUs that include both `impl/HttpClient.h` and proto headers and build on Windows: `AccountId.cc`, `RegisteredNodeAddressBookQuery.cc`, `impl/MirrorNodeGateway.cc` — comparing their include order with `FeeEstimateQuery.cc` is a quick way to confirm the diagnosis.

---

### 📋 Step-by-Step Contribution Guide

To help keep contributions consistent and easy to review, we recommend following these steps:

- [ ] Comment `/assign` to request the issue
- [ ] Wait for assignment
- [ ] Fork the repository and create a branch
- [ ] Set up the project using the instructions in `README.md`
- [ ] Make the requested changes
- [ ] Sign each commit using `-s -S`
- [ ] Push your branch and open a pull request

Read [Workflow Guide](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/docs/training/workflow.md) for step-by-step workflow guidance.
Read [README.md](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/README.md) for setup instructions.

❗ Pull requests **cannot be merged** without `S` and `s` signed commits.
See the [Signing Guide](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/docs/training/signing.md).

If you have questions while working on this issue, feel free to ask! [Hiero-SDK-C++ Discord](https://discord.com/channels/905194001349627914/1337424839761465364)

--- a/src/sdk/main/src/FeeEstimateQuery.cc
+++ b/src/sdk/main/src/FeeEstimateQuery.cc
@@ -1,6 +1,9 @@
 // SPDX-License-Identifier: Apache-2.0
-#include "FeeEstimateQuery.h"
+// Windows build requires this to be included first for some reason.
+#include <services/basic_types.pb.h> // NOLINT
+
 #include "Client.h"
+#include "FeeEstimateQuery.h"
 #include "FileAppendTransaction.h"
 #include "TopicMessageSubmitTransaction.h"
 #include "exceptions/IllegalStateException.h"

